### PR TITLE
Remove mlir-hlo (replace with stablehlo).

### PR DESCRIPTION
**Pull request by stellaraccident**
We just have to do this: I ran into an issue today where I needed to make a one line patch to stablehlo to work around a compiler issue, and it is completely unapparent how to do so given that the `mlir-hlo` repo is a read-only export and is at the tail end of a multi-week integration chain from the open-source stablehlo repo.

We've discussed this often enough and gotten +1 from everyone that they are ok with taking the e2e testing hit if it becomes necessary: It is necessary as the current situation is unmanageable.

Looking at it, I expect it wouldn't actually be very difficult to build a little runner binary out of the stablehlo interpreter and subprocess call that in order to get the testing coverage back. I leave that as an exercise to the users of this part of the stack and recommend following the breadcrumbs from the deleted `python/torch_mlir_e2e_test/stablehlo_backends/linalg_on_tensors.py` file and the `main.py` changes.

Note that I am pointing us at a stablehlo fork for the moment until it is apparent that we don't need to carry any local patches to it. We can update this in a few days if everything is clear.

### CODE REVIEW

**Review by powderluv — APPROVED** [review]

**Review by powderluv — APPROVED** [review]

**Comment by powderluv** [comment]
Thank you. Hopefully this will be the end of the index.lock issues from the MHLO repo. Hopefully we don't see it from the LLVM side though.

**Comment by stellaraccident** [comment]
> Thank you. Hopefully this will be the end of the index.lock issues from the MHLO repo. Hopefully we don't see it from the LLVM side though.

I don't know what that is coming from. If we upgrade to the style of runners that are used in IREE's PkgCI, it shouldn't be possible, fwiw.

**Review by ashay — APPROVED** [review]
LGTM, thanks for seeing this through, since it had been on the cards for a long time.

I think we could also nix the `STABLEHLO_PASS_SET` set in e2e_testing/xfail_sets.py, but we can do that in a separate commit.  Same with the examples/torchscript_stablehlo_backend_resnet.py file.  [This patch](https://github.com/llvm/torch-mlir/pull/1840/files) might be helpful to find other references to MHLO that we can now remove.

File: e2e_testing/main.py [hunk 1]
```diff
@@ -24,7 +24,6 @@
 )
```
[the remark this answers is not in the document]
Comment (stellaraccident) [reply e2e_testing/main.py:24]: Let's get both of these in a followup commit. I was going to do a deeper cut, but I'm moderately hopeful that someone steps up and puts the plumbing back in for the stablehlo interpreter (in which case, more breadcrumbs == better).

[the remark this answers is not in the document]
Comment (stellaraccident) [reply e2e_testing/main.py:24]: (I'll do an NFC cleanup after it lands)

**Comment by stellaraccident** [comment]
Ok - that is the same Windows compilation issue that drove this switch. Landing this so that I can patch it.

**Pull request merged**